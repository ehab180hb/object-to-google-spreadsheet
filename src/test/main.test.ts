import ObjectToGoogleSheet from '../index'
import { goodInputObjects, options } from './main.data'

const fakeCreds = {}
const fakeDocKey = '1gsLxL0ln8m2yJQgnLNpzvP8kGDLU1ATszwZlM'

const report = new ObjectToGoogleSheet(fakeCreds, fakeDocKey)

let result
Object.defineProperty(report, 'pushToSheet', { value: data => (result = data) })

describe('main module', () => {
  test('When optional data are given', async () => {
    await report.push(goodInputObjects, options)
    expect(result).toMatchSnapshot()
  })

  test('When optional data are not given', async () => {
    await report.push(goodInputObjects)
    expect(result).toMatchSnapshot()
  })

  test('When sorting and removing base', async () => {
    await report.push(goodInputObjects, { sort: true, removeBase: true })
    expect(result).toMatchSnapshot()
  })

  test('Auth data are passed correctly', () => {
    expect(Object.getOwnPropertyDescriptor(report, 'auth')).toMatchSnapshot()
  })

  test('Throws when given invalid input', async () => {
    // @ts-ignore
    await expect(report.push({})).rejects.toThrowErrorMatchingSnapshot()
  })
})
