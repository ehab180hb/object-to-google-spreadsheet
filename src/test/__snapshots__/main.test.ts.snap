// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main module Auth data are passed correctly 1`] = `
Object {
  "configurable": true,
  "enumerable": true,
  "value": Object {
    "creds": Object {},
    "docKey": "1gsLxL0ln8m2yJQgnLNpzvP8kGDLU1ATszwZlM",
  },
  "writable": true,
}
`;

exports[`main module Throws when given invalid input 1`] = `"Docs input must be an array"`;

exports[`main module When optional data are given 1`] = `
Object {
  "sheetName": "My Awesome Report",
  "values": Array [
    Array [
      "info",
      "Age",
      "Address",
      "Hobbies",
    ],
    Array [
      "John",
      25,
      "16 main st.",
      false,
    ],
    Array [
      "Jane",
      26,
      false,
      "swimming, Javascripting",
    ],
  ],
}
`;

exports[`main module When optional data are not given 1`] = `
Object {
  "sheetName": "New Sheet",
  "values": Array [
    Array [
      " ",
      "Age",
      "Address",
      "Hobbies",
    ],
    Array [
      "John",
      25,
      "16 main st.",
      false,
    ],
    Array [
      "Jane",
      26,
      false,
      "swimming, Javascripting",
    ],
  ],
}
`;

exports[`main module When sorting and removing base 1`] = `
Object {
  "sheetName": "New Sheet",
  "values": Array [
    Array [
      "Address",
      "Age",
      "Hobbies",
    ],
    Array [
      "16 main st.",
      25,
      false,
    ],
    Array [
      false,
      26,
      "swimming, Javascripting",
    ],
  ],
}
`;
